planner/core/common_plans.go

```diff
@@ -649,6 +649,40 @@ type SelectInto struct {
 	IntoOpt    *ast.SelectIntoOption
 }
 
+// ExplainInfoForEncode store explain info for JSON encode
+type ExplainInfoForEncode struct {
```

Comment:
Why do some fields have `omitempty` but some don't?

Reply:
> Why do some fields have `omitempty` but some don't?

Some fields in `explain` and some in `explain analyze` and some in both, some common fields don't have `omitempty`. And the `TotalMemoryConsumed` only appears when set `tidb_memory_debug_mode`.

Reply:
I mean, even for `explain`, we can choose not to print those empty fields, like empty `AccessObject`, `OperatorInfo` and `SubOperators`.

Reply:
got it,